src/types.lisp

```diff
@@ -839,22 +839,24 @@ The foreign array must be freed with foreign-array-free."
 (defmacro with-foreign-slots ((vars ptr type) &body body)
   "Create local symbol macros for each var in VARS to reference
 foreign slots in PTR of TYPE. Similar to WITH-SLOTS.
-Each var can be of the form: slot-name - in which case slot-name will
-be bound to the value of the slot or: (:pointer slot-name) - in which
-case slot-name will be bound to the pointer to that slot."
+Each var can be of the form: 
+  name                       name bound to slot of same name              
+  (:pointer name)            name bound to pointer to slot of same name
+  (name slot-name)           name bound to slot-name
+  (name :pointer slot-name)  name bound to pointer to slot-name"
   (let ((ptr-var (gensym "PTR")))
     `(let ((,ptr-var ,ptr))
        (symbol-macrolet
            ,(loop :for var :in vars
-              :collect
-              (if (listp var)
-                  (if (eq (first var) :pointer)
-                      `(,(second var) (foreign-slot-pointer
-                                       ,ptr-var ',type ',(second var)))
-                      (error
-                       "Malformed slot specification ~a; must be:`name' or `(:pointer name)'"
-                       var))
-                  `(,var (foreign-slot-value ,ptr-var ',type ',var))))
+	       :collect
+		 (if (listp var)
+		     (let ((p1 (first var)) (p2 (second var)) (p3 (third var)))
+		       (if (eq p1 :pointer)	
+			   `(,p2 (foreign-slot-pointer ,ptr-var ',type ',p2))
+			   (if (eq p2 :pointer)
+			       `(,p1 (foreign-slot-pointer ,ptr-var ',type ',p3))
+			       `(,p1 (foreign-slot-value ,ptr-var ',type ',p2)))))
+		     `(,var (foreign-slot-value ,ptr-var ',type ',var))))
```

Comment:
Oh no, tabs! Did you use a non-Emacs editor to write this? 😱

Reply:
Sorry, I attempted to make the fixes right in the browser...   Let me try again.

Reply:
OK, I think I fixed it.